x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/utils.ts

```diff
@@ -91,6 +91,26 @@ export const getNodeDocumentMode = (
   return 'na';
 };
 
+/**
+ * Checks if a node has an entity field in its documentsData.
+ * Only relevant for single-entity mode - returns false for all other modes.
+ * For single-entity nodes, checks if at least one document contains an 'entity' field.
+ */
+export const hasNodeEntityField = (node: NodeViewModel): boolean => {
```

Comment:
If we go with the new suggestion

Here you'd check documentsData. And then you'd return it's parent field.

Although I must say, if it is missing. `entity` should be the default one. 
It will also relevant to the preview flyout filters (https://github.com/elastic/kibana/issues/239954).